kyo-core/jvm/src/main/scala/kyo/StreamCompression.scala

```diff
@@ -0,0 +1,724 @@
+                        val computedCrc16 = (headerCrc32.getValue & 0xffffL).toInt
+                        val expectedCrc16 = usize16(crc16Bytes(0), crc16Bytes(1))
+                        if computedCrc16 != expectedCrc16 then
+                            Abort
+                                .fail(new StreamCompressionException("Invalid GZip header crc16"))
+                                .andThen(Loop.done(Ack.Stop))
+                        else
+                            nextState(false, 0, false)(
+                                rest,
+                                headerCrc32,
+                                emit
+                            )
+                        end if
+                    end if
+                case GunzipState.InflateInput(leftOver, inflater, contentCrc32, emit) =>
+                    if leftOver.isEmpty then
+                        Emit.runFirst(emit).map:
+                            case Present(bytes) -> emitFn =>
+                                IO(inflater.setInput(bytes.toArray))
+                                    .andThen(
+                                        Loop.continue(GunzipState.PullInflater(inflater, contentCrc32, Present(emitFn), bytes))
+                                    )
+                            case Absent -> _ =>
+                                Loop.continue(GunzipState.PullInflater(inflater, contentCrc32, Absent, Chunk.empty))
+                    else
+                        IO(inflater.setInput(leftOver.toArray))
+                            .andThen(
+                                Loop.continue(GunzipState.PullInflater(inflater, contentCrc32, Present(_ => emit), leftOver))
+                            )
+                    end if
+                case GunzipState.PullInflater(inflater, contentCrc32, maybeEmitFn, bytes) =>
+                    if inflater.finished then
+                        val leftOver = inflater.getRemaining match
+                            case remaining if remaining > 0 => bytes.takeRight(remaining)
+                            case _                          => Chunk.empty[Byte]
+                        Loop.continue(GunzipState.CheckTrailer(leftOver, inflater, contentCrc32, maybeEmitFn))
+                    else if inflater.needsInput then
+                        Emit.andMap(Chunk.empty): ack =>
+                            maybeEmitFn match
+                                case Present(emitFn) =>
+                                    Loop.continue(GunzipState.InflateInput(Chunk.empty, inflater, contentCrc32, emitFn(ack)))
+                                case Absent =>
+                                    Loop.continue(GunzipState.CheckTrailer(Chunk.empty, inflater, contentCrc32, Absent))
+                    else
+                        local.use: buffer =>
+                            Abort
+                                .catching[DataFormatException](inflater.inflate(buffer))
```

Comment:
catching `DataFormatException` forces the signature to have `DataFormatException | StreamCompressionException`, I'd like a method that can `mapError` just like ZIO